### PR TITLE
do not assume that bitnami is guanteed to be in the helm repo list

### PR DESCRIPTION
**Pull request by Nokel81**
Signed-off-by: Sebastian Malton <sebastian@malton.name>

### CODE REVIEW

**Comment by jim-docker** [comment]
I think we should be able to assume bitnami. Is this just a workaround for spectron flakiness?

**Comment by Nokel81** [comment]
No it is a work around for people (including developers) that have manually run `helm repo add ...` before the switch to `bitnami`. Namely, the code in the settings only runs that command for `bitnami` if there are no helm repos listed at all. But I think previously we did that with `stable`. I ran into this on my machine for instance.

**Comment by jim-docker** [comment]
> No it is a work around for people (including developers) that have manually run `helm repo add ...` before the switch to `bitnami`. Namely, the code in the settings only runs that command for `bitnami` if there are no helm repos listed at all. But I think previously we did that with `stable`. I ran into this on my machine for instance.

ah, I see. It would be ideal if the test was dynamic, detecting if there were repo(s) already present, and looking for bitnami if not, which is what the CI would encounter and would be a stronger test.

**Comment by Nokel81** [comment]
> > No it is a work around for people (including developers) that have manually run `helm repo add ...` before the switch to `bitnami`. Namely, the code in the settings only runs that command for `bitnami` if there are no helm repos listed at all. But I think previously we did that with `stable`. I ran into this on my machine for instance.
> 
> ah, I see. It would be ideal if the test was dynamic, detecting if there were repo(s) already present, and looking for bitnami if not, which is what the CI would encounter and would be a stronger test.

Agreed good idea. Will do that

**Comment by Nokel81** [comment]
@jim-docker Done, care to take a look again? Thanks.

File: integration/__tests__/app.tests.ts [hunk 1]
```diff
@@ -93,7 +96,10 @@ describe("Lens integration tests", () => {
       });
 
       it("ensures helm repos", async () => {
-        await app.client.waitUntilTextExists("div.repos #message-bitnami", "bitnami"); // wait for the helm-cli to fetch the bitnami repo
+        const { stdout: reposJson } = await promiseExec("helm repo list -o json");
```
Comment (jim-docker) [inline integration/__tests__/app.tests.ts:99]: better than `spawnSync()`, which is already used elsewhere in this file?

File: integration/__tests__/app.tests.ts [hunk 2]
```diff
@@ -93,7 +96,10 @@ describe("Lens integration tests", () => {
...
+        const repos = JSON.parse(reposJson);
+
+        await app.client.waitUntilTextExists("div.repos #message-bitnami", repos[0].name); // wait for the helm-cli to fetch the any repo
```
Comment (jim-docker) [inline integration/__tests__/app.tests.ts]:
```suggestion
        await app.client.waitUntilTextExists("div.repos #message-bitnami", repos[0].name); // wait for the helm-cli to fetch the repo(s)
```

**Review by jim-docker — DISMISSED** [review]

File: integration/__tests__/app.tests.ts (hunk 1 above)
Comment (Nokel81) [reply integration/__tests__/app.tests.ts:99]: We probably should be using `exec` more. See this:

https://stackoverflow.com/questions/48698234/node-js-spawn-vs-execute

**Review by jim-docker — DISMISSED** [review]

**Review by Nachasic — APPROVED** [review]

**Review by jim-docker — APPROVED** [review]

**Comment by jakolehm** [comment]
This is actually throwing errors in our pipeline.

image (2)

Should we revert?

**Comment by Nokel81** [comment]
Let me investigate, I think we can ignore this error and then test if `bitnami` is present specifically. If you want to revert then sure, but I think we should have some modification to this test.

**Pull request merged**